fw/http_msg.c

```diff
@@ -570,9 +574,8 @@ tfw_http_msg_hdr_close(TfwHttpMsg *hm)
 		 * For H2 literal headers, @parser->_hdr_tag would be set
 		 * during actual HTTP parsing.
 		 */
-		if (id < TFW_HTTP_HDR_NONSINGULAR) {
-			if (!TFW_MSG_H2(hm) || id != TFW_HTTP_HDR_COOKIE)
```

Comment:
Why the H2 case is removed? Need review from @s0nx as it was his code. @s0nx BTW I didn't get why this code is about trailer headers as stated in the comment - it seems the code is executed for all header, not trailers only.

Reply:
Previously we check that if message is not http2 we should return TFW_BLOCK. But if you look at RFS 7230 3.2.2 Note you see that we should not drop message if it has duplicate cookie header for http1 also.

Reply:
@krizhanovsky Yes, it's executed for all headers. Duplicate h2 pseudo-headers would be filtered out here, hence the comment.
@const-t raised concerns about this piece of code previously: https://github.com/tempesta-tech/tempesta/pull/1705#discussion_r1015460760